tests/forking_agent/test_agent_executor.py

```diff
@@ -330,3 +334,121 @@ def test():
     assert executor_1 is not executor_2, "Expected a different executor instance for different sources"
     assert executor_1 is not executor_3, "Expected a different executor instance for different requirements and sources"
     assert executor_2 is not executor_3, "Expected different executor instances for different requirements"
+
+
+async def test_executor_creation_and_venv_usage(pip_index: PipIndex, mpmanager_light: forking_executor.MPManager) -> None:
+    """
+    This test verifies the creation and reuse of executors based on their blueprints. It checks whether
+    the concurrency aspects and the locking mechanisms work as intended.
+    """
+
+    requirements1 = ()
+    requirements2 = ("pkg1",)
+    pip_config = PipConfig(index_url=pip_index.url)
+
+    # Prepare a source module and its hash
+    code = """
+def test():
+    return 10
+    """.encode()
+    sha1sum = hashlib.new("sha1")
+    sha1sum.update(code)
+    hv: str = sha1sum.hexdigest()
+    module_source1 = ModuleSource(
+        name="inmanta_plugins.test",
+        hash_value=hv,
+        is_byte_code=False,
+        source=code,
+    )
+    sources1 = ()
+    sources2 = (module_source1,)
+
+    blueprint1 = executor.ExecutorBlueprint(pip_config=pip_config, requirements=requirements1, sources=sources1)
+    blueprint2 = executor.ExecutorBlueprint(pip_config=pip_config, requirements=requirements2, sources=sources2)
+
+    executor_manager = mpmanager_light
+    executor_1, executor_2 = await asyncio.gather(
+        executor_manager.get_executor("agent1", "local:", code_for(blueprint1), venv_checkup_interval=0.1),
+        executor_manager.get_executor("agent2", "local:", code_for(blueprint2), venv_checkup_interval=0.1),
+    )
+
+    def retrieve_pid_agent(agent_name: str) -> int:
+        """
+        Retrieve the PID of the agent
+
+        :param agent_name: The name of the agent from which we want to retrieve the PID
+        """
+        for proc in psutil.process_iter(["pid", "name"]):
+            if f"inmanta: executor {agent_name}" in proc.name():
+                return proc.pid
```

Comment:
It's better to return the process object than to return the pid. You can then use `process.is_running()` to verify whether the process is still there. Using the pid has a race condition, because the pid might be re-used by a different process after the process stopped.